tcp.rflx

```diff
@@ -0,0 +1,129 @@
+package TCP is
+
+    -- RFC 793
+
+    type Source_Port is mod 2**16;
+    type Destination_Port is mod 2**16;
+    type Sequence_Number is mod 2**32;
+    type Acknowledgment_Number is mod 2**32;
+    type Data_Offset is range 5 .. 2**4 - 1 with Size => 4;
+
+    -- Reserved for future use. Must be zero.
+    type Reserved is range 0 .. 0 with Size => 3;
+
+    type Window is mod 2**16;
+    type Checksum is mod 2**16;
+    type Urgent_Pointer is mod 2**16;
+
+    -- Timestamp and Window_Scale options are described in RFC 7323
+    -- Selective Acknowledgment Options (SACK) and SACK_Permitted are described in RFC 2018
+    -- No_Operation, End_Of_Options_List are described in RFC 793
+    -- Maximum_Segment_Size is described in RFC 793 and RFC 879
+
+    type Kind is
+       (End_Of_Option_List     => 16#00#,
+        No_Operation           => 16#01#,
+        Maximum_Segment_Size   => 16#02#,
+        Window_Scale           => 16#03#,
+        SACK_Permitted         => 16#04#,
+        SACK                   => 16#05#,
+        Timestamps             => 16#08#)
+    with Size => 8;
+
+    type Length is range 2 .. 2**8 - 1 with Size => 8;
+    type TSval is mod 2**32;
+    type TSecr is mod 2**32;
+    type Shift_Cnt is mod 2**8;
+    type Max_Seg_Size is mod 2**16;
+    type SACK_Edges is mod 2**32;
+
+    type SACK_Data is sequence of SACK_Edges;
+
+    type Option is
+       message
+          Kind : Kind
+             then null
+                if Kind = End_Of_Option_List or Kind = No_Operation
+             then Length
+                if Kind = Maximum_Segment_Size
+                   or Kind = Window_Scale
+                   or Kind = SACK_Permitted
+                   or Kind = SACK
+                   or Kind = Timestamps;
+          Length : Length
+             then Shift_Cnt
+                if Kind = Window_Scale
+             then Max_Seg_Size
+                if Kind = Maximum_Segment_Size
+             then SACK_Data
+                if Kind = SACK
+             then TSval
+                if Kind = Timestamps and Length = 10
+             then null
+
+             -- a global variant for the message size being dependant of the field Length will be added soon
+             -- ISSUE: Componolit/RecordFlux#554
+
+                if Kind = SACK_Permitted and Message'Size = Length * 8;
+          Shift_Cnt : Shift_Cnt
```

Comment:
IMHO this is not a good place for a long comment and it should be sufficient to have `--  ISSUE: Componolit/RecordFlux#554` at each instance of `Message'Size = ...`.

Reply:
Changed in commit b040f2e19b89c3d5e955c08db220ad93426d4e19.